### DISCUSSION TITLE
Dynamic alpine x-ref not working

### DISCUSSION THREAD

**Discussion by ph7jack** [alpinejs/alpine#1601] [7. Bugs (archived)]
If i create x-ref dynamically, it dont work

This example is poor, but imagine it in a loop, x-ref="users.id" for example

Alpine Version: 3.1.0

```
<div x-data="{ name: 'Baz', init() {
    console.log(this.$refs) 
} }">
    <div x-ref="foo"></div>
    <div x-ref="bar"></div>
    <div :x-ref="`type${name}`"></div>
</div>
```

**Comment by danddanddand** [comment]
_Originally posted by @KevinBatdorf in https://github.com/alpinejs/alpine/discussions/1527#discussioncomment-879974_
